frontend/projects/upgrade/src/app/features/dashboard/segments/pages/segment-root-page/segment-root-page-content/segment-root-section-card/segment-root-section-card.component.ts

```diff
@@ -0,0 +1,169 @@
+      disabled: false,
+    },
+    {
+      name: this.translateService.instant('segments.export-all-segments.text'),
+      disabled: true,
+    },
+  ];
+
+  constructor(
+    private segmentsService: SegmentsService,
+    private translateService: TranslateService,
+    private dialogService: DialogService,
+    private tableHelpersService: CommonTableHelpersService,
+    private authService: AuthService
+  ) {}
+
+  ngOnInit() {
+    this.permissions$ = this.authService.userPermissions$;
+    this.segmentsService.fetchSegments(true);
+  }
+
+  ngAfterViewInit() {
+    // Initialize dataSource$ with segments data
+    this.dataSource$ = this.segmentsService.allSegments$.pipe(
+      map((segments: Segment[]) => {
+        const dataSource = new MatTableDataSource<Segment>(segments);
+        this.setFilterPredicateForSegments(dataSource);
+        this.applyFilter(dataSource);
+        return dataSource;
+      })
+    );
+  }
+
+  setFilterPredicateForSegments(dataSource: MatTableDataSource<Segment>) {
+    dataSource.filterPredicate = (data: Segment, filter: string) => {
+      const searchKey = this.segmentsService.selectSearchKey$;
+      let searchKeyValue: SEGMENT_SEARCH_KEY;
+
+      // Get the current search key value
+      searchKey
+        .subscribe((value) => {
+          searchKeyValue = value;
+        })
+        .unsubscribe();
+
+      filter = filter.toLowerCase();
+
+      switch (searchKeyValue) {
+        case SEGMENT_SEARCH_KEY.ALL:
+          return (
+            data.name.toLowerCase().includes(filter) ||
+            data.status.toLowerCase().includes(filter) ||
+            data.context.toLowerCase().includes(filter)
+          );
+        case SEGMENT_SEARCH_KEY.NAME:
+          return data.name.toLowerCase().includes(filter);
+        case SEGMENT_SEARCH_KEY.STATUS:
+          return data.status.toLowerCase().includes(filter);
+        case SEGMENT_SEARCH_KEY.CONTEXT:
+          return data.context.toLowerCase().includes(filter);
+        default:
+          return data.name.toLowerCase().includes(filter);
+      }
+    };
+  }
+
+  applyFilter(dataSource: MatTableDataSource<Segment>) {
+    let searchString: string;
+
+    // Get the current search string value
+    this.searchString$
+      .subscribe((value) => {
+        searchString = value;
+      })
+      .unsubscribe();
+
+    if (searchString) {
+      dataSource.filter = searchString.trim().toLowerCase();
+    }
+  }
+
+  onSearch(params: CommonSearchWidgetSearchParams<SEGMENT_SEARCH_KEY>) {
+    this.segmentsService.setSearchString(params.searchString);
+    this.segmentsService.setSearchKey(params.searchKey as SEGMENT_SEARCH_KEY);
+  }
+
+  onAddSegmentButtonClick() {
+    // this.dialogService.openNewSegmentModal();
+  }
+
+  onMenuButtonItemClick(menuButtonItemName: string) {
+    if (menuButtonItemName === 'Import Segment') {
+      // this.dialogService.openImportSegmentsModal();
+    } else if (menuButtonItemName === 'Export All Segments') {
```

Comment:
are we going to do export all segments?

Reply:
It's just a placeholder but we can remove that.